src/test/java/org/opensearch/integTest/VectorDBToolIT.java

```diff
@@ -0,0 +1,204 @@
+        deleteExternalIndices();
+        deleteModel(modelId);
+    }
+
+    public void testVectorDBToolInFlowAgent() {
+        String agentId = createAgent(registerAgentRequestBody);
+
+        // match similar text, doc1 match with higher score
+        String result = executeAgent(agentId, "{\"parameters\": {\"question\": \"c\"}}");
+        assertEquals(
+            "The agent execute response not equal with expected.",
+            "{\"_index\":\"test_index\",\"_source\":{\"text\":\"hello world\"},\"_id\":\"0\",\"_score\":0.7046764}\n"
+                + "{\"_index\":\"test_index\",\"_source\":{\"text\":\"a b\"},\"_id\":\"1\",\"_score\":0.2649903}\n",
+            result
+        );
+
+        // match exact same text case, doc0 match with higher score
+        String result1 = executeAgent(agentId, "{\"parameters\": {\"question\": \"hello\"}}");
+
+        assertEquals(
+            "The agent execute response not equal with expected.",
+            "{\"_index\":\"test_index\",\"_source\":{\"text\":\"hello world\"},\"_id\":\"0\",\"_score\":0.56714886}\n"
+                + "{\"_index\":\"test_index\",\"_source\":{\"text\":\"a b\"},\"_id\":\"1\",\"_score\":0.24236833}\n",
+            result1
+        );
+
+        // if blank input, call onFailure and get exception
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(
+                exception.getMessage(),
+                allOf(containsString("[input] is null or empty, can not process it."), containsString("IllegalArgumentException"))
+            );
+    }
+
+    public void testVectorDBToolInFlowAgent_withIllegalSourceField_thenGetEmptySource() {
+        String agentId = createAgent(registerAgentRequestBody.replace("text", "text2"));
+        String result = executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}");
+        assertEquals(
+            "The agent execute response not equal with expected.",
+            "{\"_index\":\"test_index\",\"_source\":{},\"_id\":\"0\",\"_score\":0.70493275}\n"
+                + "{\"_index\":\"test_index\",\"_source\":{},\"_id\":\"1\",\"_score\":0.2650575}\n",
+            result
+        );
+    }
+
+    public void testVectorDBToolInFlowAgent_withIllegalEmbeddingField_thenThrowException() {
+        String agentId = createAgent(registerAgentRequestBody.replace("\"embedding\"", "\"embedding2\""));
+        Exception exception = assertThrows(ResponseException.class, () -> executeAgent(agentId, "{\"parameters\": {\"question\": \"a\"}}"));
+
+        org.hamcrest.MatcherAssert
+            .assertThat(
+                exception.getMessage(),
+                allOf(containsString("all shards failed"), containsString("SearchPhaseExecutionException"))
+            );
+    }
+
+    public void testNeuralSparseSearchToolInFlowAgent_withIllegalIndexField_thenThrowException() {
```

Comment:
should rename the NeuralSparseSearchTool to VectorDBTool here

Reply:
same as below

Reply:
right, thx